CHANGELOG.md

```diff
@@ -7,6 +7,16 @@ More expansive patch notes and explanations may be found in the specific [pathfi
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## Unreleased
+
+### Added
+
+- Added `x-request-id` header to RPC responses. If the request does not have the header set then an ID is generated. This can be used to identify a specific caller's request/response within the node's logs. 
```

Comment:
Slight concern: duplicated IDs will happen. If there is an easy way to distinguish between generated vs user-provided we should consider it, or maybe just mention this explicitly here?

Reply:
I think its a bad idea to tamper with the header if it was specified. I don't think there is much of an alternative?

Reply:
We could make ours more complicated than just a number I suppose, but I'm not really sure we want to do that? Could use a GUID for example..

Reply:
We could simply add a prefix, or just say duplicated IDs may happen.

Reply:
Done :)